File: class2relational/src/main/java/org/atlanmod/model/dynamic/classModel/ClassAttribute.java
package org.atlanmod.model.dynamic.classModel;

public class ClassAttribute extends ClassElement {

    public ClassAttribute(String id, String name){
        super(ClassMetamodel.ATTRIBUTE);
        super.eSetProperty("id", id);
        super.eSetProperty("name", name);
        super.eSetProperty("derived", false);
        super.eSetProperty("multivalued", false);
    }

    public ClassAttribute(String id, String name, Boolean derived, Boolean multivalued){
        super(ClassMetamodel.ATTRIBUTE);
        super.eSetProperty("id", id);
        super.eSetProperty("name", name);
        super.eSetProperty("derived", derived);
        super.eSetProperty("multivalued", multivalued);
    }

    public String getId(){
        return (String) super.eGetProperty("id");
    }

    public void setId(String id){
        super.eSetProperty("id", id);
    }

    public String getName(){
        return (String) super.eGetProperty("name");
    }

    public void setName(String name){
        super.eSetProperty("name", name);
    }

    public Boolean isDerived(){
        return (Boolean) super.eGetProperty("derived");
    }

    public void setDerived(Boolean derived){
        super.eSetProperty("derived", derived);
    }

    public Boolean isMultivalued(){
        return (Boolean) super.eGetProperty("multivalued");
    }

    public void setMultivalued(Boolean multivalued){
        super.eSetProperty("multivalued", multivalued);
    }

    public ClassClass getClass_(){
        return (ClassClass) super.eGetReference("class");
    }

    public void setClass_(ClassClass c){
        super.eSetReference("class", c);
    }

    @Override
    public String toString() {
        return super.getType() +
                "(" +
                "id:" + this.getId() +
                ", name:" + this.getName() +
                ", derived:" + this.isDerived() +
                ")";
    }
}

File: relational2class/src/main/scala/org/atlanmod/model/classmodel/ClassAttribute.scala
package org.atlanmod.model.classmodel

import org.atlanmod.model.IdGenerator

class ClassAttribute extends ClassElement(ClassMetamodel.ATTRIBUTE) {

    def this(name: String){
        this()
        super.eSetProperty("id", IdGenerator.id())
        super.eSetProperty("name", name)
        super.eSetProperty("multivalued", false)
    }

    def this(name: String, multi: Boolean){
        this()
        super.eSetProperty("id", IdGenerator.id())
        super.eSetProperty("name", name)
        super.eSetProperty("multivalued", multi)
    }

    def this(id: String, name: String, multi: Boolean){
        this()
        super.eSetProperty("id", id)
        super.eSetProperty("name", name)
        super.eSetProperty("multivalued", multi)
    }

    def this(id: String, name: String){
        this()
        super.eSetProperty("id", id)
        super.eSetProperty("name", name)
        super.eSetProperty("multivalued", false)
    }

    override def getId: String = super.eGetProperty("id").asInstanceOf[String]
    def getName: String = super.eGetProperty("name").asInstanceOf[String]
    def setName(name: String): Unit = super.eSetProperty("name", name)
    def isMultivalued: Boolean = super.eGetProperty("multivalued").asInstanceOf[Boolean]
    def setMultivalued(multi: Boolean): Unit = super.eSetProperty("multivalued", multi)

    override def toString: String =
        getType + "([" + getId + "] " + getName + (if(isMultivalued) "multivalued)" else ")")

    override def weak_equals(o: Any): Boolean = {
        o match {
            case obj: ClassAttribute =>
                this.getName.equals(obj.getName) && this.isMultivalued.equals(obj.isMultivalued)
            case _ => false
        }
    }

}
